### ISSUE TITLE
object-curly-spacing: Add option for different styles in import/export

### ISSUE THREAD

**Issue by RecuencoJones** [eslint/eslint#9973]
**What rule do you want to change?** 

`object-curly-spacing`

**Does this change cause the rule to produce more or fewer warnings?** 

More when option is used

**How will the change be implemented? (New option, new default behavior, etc.)?** 

New items for second option:

- `import`: enforce space style inside import braces when inline 
- `export`: enforce space style inside export braces when inline

Maybe it could just be one option `importExport`

**Please provide some example code that this change will affect:**

```js
/*eslint object-curly-spacing: ["error", "never", { "import": "always", "export": "always" }]*/
import { foo } from './foo';

const bar = {foo, values: [1,2,3,4]};

export { bar };
```

**What does the rule currently do for this code?**

```
1:8 - There should be no space after '{'. (object-curly-spacing)
1:14 - There should be no space before '}'. (object-curly-spacing)
5:8 - There should be no space after '{'. (object-curly-spacing)
5:14 - There should be no space before '}'. (object-curly-spacing)
```

See example [here](https://eslint.org/demo/#eyJ0ZXh0IjoiaW1wb3J0IHsgZm9vIH0gZnJvbSAnLi9mb28nO1xuXG5jb25zdCBiYXIgPSB7Zm9vLCB2YWx1ZXM6IFsxLDIsMyw0XX07XG5cbmV4cG9ydCB7IGJhciB9OyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJvYmplY3QtY3VybHktc3BhY2luZyI6Mn0sImVudiI6eyJlczYiOnRydWV9fX0=)

**What will the rule do after it's changed?**

It will correctly warn about different styles for import/export and object literals.

**Comment by not-an-aardvark** [comment]
Hi, thanks for the proposal. At first glance I'm not sure whether it would be worthwhile to add an option for this, because I'm not sure this is a common preference. However, I'd be open to supporting it if there are a lot of users that want to enforce this style.

In the meantime, keep in mind that you could also implement this as a [custom rule](https://eslint.org/docs/developer-guide/working-with-rules) for your project regardless of whether it gets added to the bundled `object-curly-spacing` rule.

**Comment by iarroyo5** [comment]
Looks like this rule is being requested for a while and I think it would be worthy its implementation, https://github.com/eslint/eslint/issues/7361 & https://github.com/eslint/eslint/pull/8671 are relational  discussions about this feature

**Comment by ackvf** [comment]
When asked [here](https://github.com/airbnb/javascript/issues/1407), one airbnb engineer stated:

Q1
> Hi airbnb team,
> Can you give us your thinking behind applying the object-curly-spacing rule?

Q2
>I would be also interested in this question:
In similar situations, e.g. the function definition foo(b, a = 2), or array definition [0, 1, 2] it is the other way around. I would think that if its better for readability in one place, then in the other too... or is there another reason?

A
>In both those cases, they're **lists of items**; the comma is sufficient to separate them.
>In the case of objects, they're **key/value pairs**; the spacing helps avoid confusion here.

However, given destructuring assignment and import/exports, this statement is not completely true, also Airbnb itself contains inconsistency in their code base as in this example:
[[5.3]](https://github.com/airbnb/javascript#destructuring--object-over-array)

Simplified and emphasized
```
function returnArray() {
  ...
  return [top, right, bottom, left];
}

function returnObject() {
  ...
  return { top, right, bottom, left };
}

const [top, right] = returnArray();
const { top, right } = returnObject();
```
Notice that they are syntactically and semantically identical statements, but both with different code style.


I would agree that **list of items** and **key:value pairs** are different, but insufficient to describe all these language constructs: `arrays, objects, destructuring and import/exports`.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
